test/smoke/framework/testCore/src/main/java/com/microsoft/applicationinsights/smoketest/AiSmokeTest.java

```diff
@@ -421,13 +540,49 @@ public void resetMockedIngestion() throws Exception {
 
 	@AfterWithParams
 	public static void tearDownContainer(final String appServer, final String os, final String jreVersion) throws Exception {
```

Comment:
It could, but I didn't want to over complicate this during the first phase. If performance needs to be improved, this is something we can investigate. At this point, I don't think the smoke test perf is too bad.

Reply:
There is some order that needs to be preserved:
1. Start dependencies (assume no inter-dependencies; otherwise it would need to be serial)
2. Start test app
3. Do Test
4. Stop test app
5. Stop dependencies

As long as there are no inter-dependencies, the order of stopping dependencies is arbitrary; but I use a stack so everything is stopped in the reverse order of it being started.